### PR TITLE
Get group by group_id & list groups

### PR DESCRIPTION
**Pull request by carrala**
Tests pass for both these parts.

### CODE REVIEW

File: __tests__/storage.test.js [hunk 1]
```diff
@@ -213,13 +213,46 @@ describe('Tozny', () => {
     const deleted = await ops.deleteAnonymousNote(written.noteId, signingPair)
     expect(deleted).toBe(true)
   })
-  it('can create groups', async () => {
+  it('can create, read and list groups', async () => {
```
Comment (galxy25) [inline __tests__/storage.test.js:216]: Make each of these a separate test, generally we want each test to be scoped to testing a single higher level concept, that way if the test fails its clear exactly what functionality of the api is broken / has regressed.

**Review by galxy25 — APPROVED** [review]
Approved pending splitting out of mega test into individual test cases

File: lib/storage/client.js [hunk 2]
```diff
@@ -1106,6 +1107,69 @@ class Client extends CryptoConsumer {
...
+    let request = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupID}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let storedGroup = await checkStatus(request)
```
Comment (lkwdwrd) [inline lib/storage/client.js:1128]: nit: we can use the helper which combines the status check/json decode here

File: lib/storage/client.js [hunk 3]
```diff
@@ -1106,6 +1107,69 @@ class Client extends CryptoConsumer {
...
+    nextToken = null,
+    max = null
+  ) {
+    const urlData = {
+      client_id: clientID,
+      group_names: groupNames,
+      nextToken: nextToken,
+      max: max,
+    }
+    let encodedUrl = urlEncodeData(urlData)
```
Comment (lkwdwrd) [inline lib/storage/client.js:1155]: huh... I think this helper was actually written to url-encode POST data in request bodies, but this is a great use for it. Nice!

File: types/group.js [hunk 4]
```diff
@@ -6,7 +6,9 @@ class Group extends Serializable {
     super()
     this.groupName = data.groupName
     this.publicKey = membershipKeys.publicKey
-    this.encryptedGroupKey = membershipKeys.encryptedGroupKey
+    if (membershipKeys.encryptedGroupKey !== null) {
```
Comment (lkwdwrd) [inline types/group.js:9]: is there a reason to not allow this as a null value?

File: lib/storage/client.js [hunk 5]
```diff
@@ -1106,6 +1107,69 @@ class Client extends CryptoConsumer {
...
+    let request = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups?${encodedUrl}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let storedGroups = await checkStatus(request)
```
Comment (lkwdwrd) [inline lib/storage/client.js:1165]: combined helper again :)

**Review by lkwdwrd — APPROVED** [review]
Some very minor nits to consider and a question. This looks good!

**Pull request merged**